stdlib/src/os/os.mojo

```diff
@@ -328,3 +328,60 @@ fn unlink[pathlike: os.PathLike](path: pathlike) raises:
 
     """
     remove(path.__fspath__())
+
+
+# ===----------------------------------------------------------------------=== #
+# mkdir/rmdir
+# ===----------------------------------------------------------------------=== #
+
+
+fn mkdir(path: String, mode: Int = 0o777) raises:
+    """Creates a directory at the specified path.
+    If the directory can not be created an error is raised.
+    Absolute and relative paths are allowed, relative paths are resolved from cwd.
+
+    Args:
+      path: The path to the directory.
+      mode: The mode to create the directory with.
+    """
+
+    var error = external_call["mkdir", Int](path._as_ptr(), mode)
```

Comment:
Should have commented to this pr instead (btw thanks for splitting it up) but the result should be Int32 here

Reply:
Sure, could you just confirm that I should use Int32?, in an older PR I was told to use Int: https://github.com/modularml/mojo/pull/2310#discussion_r1569653405

Then I could do the change in `os.remove`  in this pulll if it is ok to merge both changes a the same time

Reply:
Yes. the unlink should be Int32